src/stream/src/executor/error.rs

```diff
@@ -53,6 +50,9 @@ enum StreamExecutorErrorInner {
     #[error("Failed to align barrier: expected {0:?} but got {1:?}")]
     AlignBarrier(Box<Barrier>, Box<Barrier>),
 
+    #[error("Connector error: {0}")]
+    ConnectorError(String),
```

Comment:
Can we use a `BoxError` here, so that we can both store `ConnectorError` and `anyhow::Error`.

Reply:
Done

Reply:
Done @BugenZhao, please could you help review?